File: onlineshop4dvds_api/src/utils/genre-type.ts
export enum GenreType {
    Music = 0,
    Movie = 1,
    Game = 2
}

export const toString = (genreType: GenreType) => {
    switch (genreType) {
        case GenreType.Game:
            return "Game";
        case GenreType.Movie:
            return "Movie";
        case GenreType.Music:
            return "Album";
        default:
            throw new Error("Invalid genre type");
    }
}

File: onlineshop4dvds_api/Utils/GenreType.cs
using System.Runtime.CompilerServices;

namespace OnlineShop4DVDS.Utils;

public enum GenreType
{
    Music,
    Movie,
    Game
}

public static class GenreTypeExtensions 
{
    public static string ToStringType(this GenreType genreType)
    {
        return genreType switch
        {
            GenreType.Music => "Album",
            GenreType.Game => "Game",
            GenreType.Movie => "Movie",
            _ => throw new SwitchExpressionException(genreType)
        };
    }
}
